File: plugins/elixir-kit/testdata/good/lib/ok.ex
# Good fixture: to_existing_atom, parameterized query, documented allow.
defmodule Ok do
  def atom_ok(raw), do: String.to_existing_atom(raw)

  def sql_ok(repo, id), do: repo.query("SELECT id FROM users WHERE id = $1", [id])

  # Documented intentional seam (boot probe); keep allow on the smell line.
  def documented_legacy(raw), do: String.to_atom(raw) # elixir-rg-allow: fixture documents allow marker for intentional atom seam
end

File: plugins/php-kit/testdata/good/src/ok.php
<?php
// Good fixture: binds, allowlisted unserialize, documented allow.

function find_ok(PDO $pdo, int $id): ?array
{
    $stmt = $pdo->prepare('SELECT id, name FROM users WHERE id = ?');
    $stmt->execute([$id]);
    $row = $stmt->fetch(PDO::FETCH_ASSOC);

    return $row === false ? null : $row;
}

function decode_ok(string $raw): array
{
    return json_decode($raw, true, 512, JSON_THROW_ON_ERROR);
}

// Documented intentional seam (boot probe); keep allow on the smell line.
function documented_legacy(string $code)
{
    return eval($code); // php-rg-allow: fixture documents allow marker for intentional eval seam
}
